crm_claim_rma/models/crm_claim.py

```diff
@@ -0,0 +1,210 @@
+                line = {
+                    'name': invoice_line.name,
+                    'claim_origin': "none",
+                    'invoice_line_id': invoice_line.id,
+                    'product_id': invoice_line.product_id.id,
+                    'product_returned_quantity': invoice_line.quantity,
+                    'unit_sale_price': invoice_line.price_unit,
+                    'location_dest_id': location_dest.id,
+                    'state': 'draft',
+                }
+                line.update(warranty_values(invoice_line.invoice_id,
+                                            invoice_line.product_id))
+                claim_lines.append((0, 0, line))
+
+            value = self._convert_to_cache(
+                {'claim_line_ids': claim_lines}, validate=False)
+            self.update(value)
+
+        if self.invoice_id:
+            self.delivery_address_id = self.invoice_id.partner_id.id
+
+    @api.model
+    def message_get_reply_to(self, res_ids, default=None):
+        """ Override to get the reply_to of the parent project.
+        """
+        results = dict.fromkeys(res_ids, default or False)
+        if res_ids:
+            claims = self.browse(res_ids)
+            results.update({
+                claim.id: self.env['crm.team'].message_get_reply_to(
+                    [claim.team_id], default
+                )[claim.team_id] for claim in claims if claim.team_id
+            })
+
+        return results
+
+    @api.multi
+    def message_get_suggested_recipients(self):
+        recipients = super(CrmClaim, self).message_get_suggested_recipients()
+        try:
+            for claim in self:
+                if claim.partner_id:
+                    claim._message_add_suggested_recipient(
+                        recipients,
+                        partner=claim.partner_id,
+                        reason=_('Customer')
+                    )
+                elif claim.email_from:
+                    claim._message_add_suggested_recipient(
+                        recipients,
+                        email=claim.email_from,
+                        reason=_('Customer Email')
+                    )
+        except exceptions.AccessError:
+            # no read access rights -> just ignore suggested recipients
+            # because this imply modifying followers
+            pass
+        return recipients
+
+    def _get_sequence_number(self, code_id):
+        claim_type_code = self.env['crm.claim.type'].\
+            browse(code_id).ir_sequence_id.code
+        sequence = self.env['ir.sequence']
+
+        return claim_type_code and sequence.next_by_code(
+            claim_type_code
+        ) or '/'
+
+    @api.model
+    def create(self, values):
+        values = values or {}
+        if 'code' not in values or not values.get('code') \
+                or values.get('code') == '/':
+            values['code'] = self._get_sequence_number(values['claim_type'])
```

Comment:
You can have a `KeyError` if 'claim_type' is not in values (it is not filled with the default value) ; it should be tested, and retrieved with the default method if not available.